server/routerlicious/packages/services-core/src/mongoCheckpointRepository.ts

```diff
@@ -62,6 +62,22 @@ export class MongoCheckpointRepository implements ICheckpointRepository {
 		documentId: string,
 		tenantId: string,
 	): { _id: string; documentId: string } & any {
+		const isError = !documentId || !tenantId ? true : false;
+
+		if (isError) {
+			const error = new Error(`Cannot create filter due to missing parameter`);
+			Lumberjack.error(
+				"Missing parameter when writing checkpoint.",
+				{
+					...getLumberBaseProperties(documentId, tenantId),
+					documentIdInput: documentId,
+					tenantIdInput: tenantId,
+				},
+				error,
+			);
+			throw error;
```

Comment:
Do we want to throw? Will throw here affect existing flows?

Reply:
I think for now while we analyze this we can just log the error. Eventually, we should throw the error. We catch any errors thrown within the checkpointRepository on the checkpointService level, and this would prevent us from trying a db operation that will fail due to an incomplete/empty filter